### PR TITLE
Delete students from an event added

### PR DESCRIPTION
**Pull request by qu-bit1**

### CODE REVIEW

**Comment by 1-Harshit** [comment]
Implement un-recruit logic here as well


---
Harshit Raj
Junior Undergraduate
CSE, IIT Kanpur
1-harshit.github.io<https://1-harshit.github.io>
________________________________
From: Sagar Arora ***@***.***>
Sent: Sunday, August 13, 2023 8:25:27 PM
To: spo-iitk/ras-backend ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [spo-iitk/ras-backend] Delete students from an event added (PR #137)

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/spo-iitk/ras-backend/pull/137

Commit Summary

  *   ca93345<https://github.com/spo-iitk/ras-backend/pull/137/commits/ca9334571053fda5efe7efd3c0895924c57b52eb> Delete students from an event added

File Changes

(3 files<https://github.com/spo-iitk/ras-backend/pull/137/files>)

  *   M application/admin.student.go<https://github.com/spo-iitk/ras-backend/pull/137/files#diff-96ac896d19db9984e4fee7909931ce7aa1682be2d18c44df31f0e890a488af9f> (33)
  *   M application/db.student.go<https://github.com/spo-iitk/ras-backend/pull/137/files#diff-ed1321f37f496989962fec4ddaeeba6963c02d7363d60ab3baf53ef7eff37da7> (10)
  *   M application/router.go<https://github.com/spo-iitk/ras-backend/pull/137/files#diff-646905c424536880e6bc474a3c09e8dd7b1ecc9178dcc7c1e1064c7e07d35267> (2)

Patch Links:

  *   https://github.com/spo-iitk/ras-backend/pull/137.patch
  *   https://github.com/spo-iitk/ras-backend/pull/137.diff

File: application/db.student.go [hunk 1]
```diff
@@ -35,3 +35,13 @@ func getCurrentApplicationCount(ctx *gin.Context, sid uint) (int, error) {
 		Group("company_recruitment_cycle_id").Count(&count)
 	return int(count), tx.Error
 }
+
+func deleteStudentByID(ctx *gin.Context, studentID uint) error {
```
Comment (1-Harshit) [inline application/db.student.go:39]: Deletestedenteventbyid

Comment (1-Harshit) [reply application/db.student.go:39]: Oh lmao, do you know what this will do?

File: application/db.student.go [hunk 2]
```diff
@@ -35,3 +35,13 @@ func getCurrentApplicationCount(ctx *gin.Context, sid uint) (int, error) {
...
+	tx := db.WithContext(ctx).Where("student_recruitment_cycle_id = ?", studentID).Delete(&EventStudent{})
+	return tx.Error
+}
+
+func deleteAllStudents(ctx *gin.Context, eventID uint, students *[]EventStudent) error {
```
Comment (1-Harshit) [inline application/db.student.go:44]: Again delete event student by event id

**Review by 1-Harshit — COMMENTED** [review]
Think before you commit

**Comment by 1-Harshit** [comment]
Np, happens. Fix in this PR only.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by qu-bit1** [comment]
closing this PR and creating a new one as there were some conflicts and required some new changes

**Pull request closed without merging**